### DISCUSSION TITLE
How to test unit with jest in typescript but without babel.

### DISCUSSION THREAD

**Discussion by Arslanmajeed270** [vercel/vercel#5529] [Help]
Hi,
I am trying to do unit testing with jest. But the problem is that i am unable to do that because there is not a single clue which could be helpful. All the example found on the google used babel but now NextJS has typescript support. So anyone could help me how to do unit test with typescript but must be without babel config.

Kind Regards,
Arslan

**Comment by dmudro** [comment]
There should be plenty of examples of Next.js + TS + Jest online including [this one in the Next.js repo](https://github.com/vercel/next.js/tree/canary/examples/with-typescript-eslint-jest).

**Reply by AlexJeffcott** [reply]
that uses 'babel-jest' for the transform in the jest config.
```
transform: {
    '^.+\\.(ts|tsx)$': 'babel-jest'
```
